lib/java_buildpack/framework/app_dynamics_agent.rb

```diff
@@ -83,7 +84,7 @@ def host_name(java_opts, credentials)
 
       def node_name(java_opts, credentials)
         name = credentials['node-name'] || @configuration['default_node_name']
-        java_opts.add_system_property('appdynamics.agent.nodeName', "#{name}")
+        java_opts.add_system_property('appdynamics.agent.nodeName', @appdynamics_tier_name + ":#{name}")
```

Comment:
@pranta @nebhale So, in our app dynamics service broker we set the node name credential to an expression like:
```
$(expr "$VCAP_APPLICATION" : '.*"application_name[": ]*\\([a-z_0-9-]\\+\\).*')[$(expr "$VCAP_APPLICATION" : '.*"instance_index[": ]*\\([0-9]\\+\\).*')]-[provo]
```

So our node looks like "cf_app_name[0]-[deployment-name]"

I agree with @nkatre that index alone is not a good default for node name.  However, I don't like the idea of hard coding in the BP the prefixing of the Tier name in the Node name.

Why not change the default node name to include the application name similar to the expression I'm using?  Then people can continue to make the node name whatever they want.

Am I following this discussion correctly?